src/twisted/protocols/sip.py

```diff
@@ -543,59 +583,75 @@ class MessagesParser(basic.LineReceiver):
     version = "SIP/2.0"
     acceptResponses = 1
     acceptRequests = 1
-    state = "firstline" # or "headers", "body" or "invalid"
+    state = "firstline" # Or "headers", "body" or "invalid"
 
     debug = 0
 
     def __init__(self, messageReceivedCallback):
         self.messageReceived = messageReceivedCallback
         self.reset()
 
+
     def reset(self, remainingData=""):
         self.state = "firstline"
-        self.length = None # body length
-        self.bodyReceived = 0 # how much of the body we received
+        self.length = None # Body length
+        self.bodyReceived = 0 # How much of the body we received
         self.message = None
         self.header = None
         self.setLineMode(remainingData)
 
+
     def invalidMessage(self):
         self.state = "invalid"
         self.setRawMode()
 
+
     def dataDone(self):
-        # clear out any buffered data that may be hanging around
+        """
+        Clear out any buffered data that may be hanging around.
+        """
         self.clearLineBuffer()
         if self.state == "firstline":
             return
         if self.state != "body":
             self.reset()
             return
         if self.length == None:
-            # no content-length header, so end of data signals message done
+            # No content-length header, so end of data signals message done
             self.messageDone()
         elif self.length < self.bodyReceived:
-            # aborted in the middle
+            # Aborted in the middle
             self.reset()
         else:
-            # we have enough data and message wasn't finished? something is wrong
+            # We have enough data and message wasn't finished? something is wrong
             raise RuntimeError("this should never happen")
 
+
     def dataReceived(self, data):
         try:
+            if isinstance(data, unicode):
```

Comment:
Clients outside of Twisted can be written with things like dataReceived("someString")
so that means in Py2 it is bytes, but in Py3 it is unicode.  So we need this conversion.